src/profile/components/additionalInformation/AdditionalInformation.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { Form, Formik, FormikProps } from 'formik';
+import to from 'await-to-js';
+
+import profileConstants from '../../constants/profileConstants';
+import commonFormStyles from '../../../common/cssHelpers/form.module.css';
+import ProfileSection from '../../../common/profileSection/ProfileSection';
+import FormikDropdown, {
+  OptionType,
+  HdsOptionType,
+} from '../../../common/formikDropdown/FormikDropdown';
+import { useProfileDataEditor } from '../../hooks/useProfileDataEditor';
+import {
+  additionalInformationType,
+  AdditionalInformationValue,
+} from '../../helpers/editData';
+import useNotificationContent from '../editingNotifications/useNotificationContent';
+import EditingNotifications from '../editingNotifications/EditingNotifications';
+
+type FormValues = {
+  language: AdditionalInformationValue['language'];
+};
+
+function AdditionalInformation(): React.ReactElement | null {
+  const { editDataList, save } = useProfileDataEditor({
+    dataType: additionalInformationType,
+  });
+  const {
+    content,
+    setErrorMessage,
+    setSuccessMessage,
+    clearMessage,
+  } = useNotificationContent();
+  const { t } = useTranslation();
+  if (!editDataList || !editDataList[0]) {
+    return null;
+  }
+  const editData = editDataList[0];
+  const { value } = editData;
+  const { language } = value as AdditionalInformationValue;
+  const languageOptions: OptionType[] = profileConstants.LANGUAGES.map(
+    languageOption => ({
+      value: languageOption,
+      label: t(`LANGUAGE_OPTIONS.${languageOption}`),
+    })
+  );
+
+  const updateLanguage = async (newLanguage: FormValues['language']) => {
+    clearMessage();
+    const [error] = await to(
+      save(editData, { language: newLanguage } as AdditionalInformationValue)
+    );
+    if (error) {
+      setErrorMessage('save');
+    } else {
+      setSuccessMessage('save');
+    }
+  };
+
+  return (
+    <ProfileSection>
+      <h3 className={commonFormStyles.sectionTitle}>
+        {t('profileForm.language')}
+      </h3>
+      <div className={commonFormStyles.multiItemWrapper}>
+        <Formik
+          initialValues={{ language }}
+          onSubmit={() => {
+            /* onChange does this. This comment is here because onSubmit is mandatory and function cannot be empty */
```

Comment:
Kind of weird, that every other change in the data requires a "start modifying/modify/save" cycle, but this one field is saved immediately :thinking:

Reply:
Also, IMO this leads to a bit sub-par UX in error situations. With the current solution, when one selects another language, but the saving fails, the newly selected language remains selected, and an error is shown which asks to "try again". How does the user try again when there's no save button? The user has to temporarily select a third language, then reselect the one they originally wanted to select and hope that this time the saving succeeds.

I wonder if this is the best flow for the user...

Reply:
Good point and clearly a bug. The selected value is now reverted when an error occurs, so change can be made again without doing any dummy changes.

IMO, this is much more convenient for the user than rendering first:

"Profile language: English [edit]"

And then

"Profile language: [English ˆ]"
[Save] [Cancel]

Someday when there is more Additional information than just language dropdown, edit mode could be used. 

Changing a primary / removing an item does not require edit mode either.

Reply:
> Changing a primary / removing an item does not require edit mode either.

Very good point, somehow I didn't realize this earlier. This is good now.